### PR TITLE
BL-314 refactored courses by id endpoint

### PR DESCRIPTION
**Pull request by luisbryandr**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Loom Video

(https://www.loom.com/share/b38d602e0d324e1cb7ffacb053a6f708)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have removed unnecessary comments/console logs from my code
- [X] I have made corresponding changes to the documentation if necessary (optional)
- [ X] My changes generate no new warnings
- [ X] I have checked my code and corrected any misspellings
- [ X] No duplicate code left within changed files
- [ X] Size of pull request kept to a minimum
- [ X] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by ShanaeL29 — APPROVED** [review]
Nice catch recognizing a piece of repetitive code and removing it to reduce database calls. Even a simple change can make a big enough difference in the final product. Continue to pay good attention to detail and review your team's work as this is a great skill to have in this industry.

**Review by cavazosgeorge — CHANGES_REQUESTED** [review]
Hey @luisbryandr great job noticing the redundant API calls within this endpoint. I wrote this endpoint initially and learned something from your loom explanation. 

Don't forget to write a description explaining the overview of this change and why it was needed!

**Review by JorgeEvangelista500 — APPROVED** [review]
Great job changing the documentation, maybe adding a catch to stay consistent with other endpoints

File: README.md [hunk 1]
```diff
@@ -13,24 +13,26 @@
...
-| [PUT]    | /profile/                   | Returns an event object with the specified `okta`. Updates specific profile.                                             |
-| [DELETE] | /profile/:okta_id/             | Returns an event object with the specified `okta`. Deletes specific profile.                                             |
+| [GET]    | /profile/role/:role_id <b>(BUG: does not return any data)</b>                                                          | Returns an array filled with event objects that contain information based on role_id for all profiles of a role_id type. |
+| [POST]   | /profile/                                                                                                              | Requires a name, password, and email. Registers a new user.                                                              |
+| [PUT]    | /profile/                                                                                                              | Returns an event object with the specified `okta`. Updates specific profile.                                             |
```
Comment (JessWillCode) [inline README.md:24]: These look almost identical, so confused why it's showing that it was removed and then added back.

File: api/courses/coursesRouter.js [hunk 2]
```diff
@@ -225,19 +225,8 @@ router.get('/', authRequired, function (req, res, next) {
...
-  const { course_id } = req.course;
-  try {
-    Courses.findByCourseId(course_id).then((course) => {
-      res.status(200).json(course);
-    });
-  } catch (err) {
-    next(err);
-  }
+router.get('/:course_id', authRequired, checkCourseExists, (req, res) => {
+  res.status(200).json(req.course);
```
Comment (JessWillCode) [inline api/courses/coursesRouter.js:229]: Nice job refactoring the code down! This is so much cleaner than having the model function inside.

**Review by JessWillCode — APPROVED** [review]
Nice Refactor! I think that the ReadME is easier to read, and the function in the code is so much cleaner to read and understand. Thanks for this!

**Review by EduarddoEscobar — APPROVED** [review]
Great work making the code more DRY!

**Review by Ike888 — APPROVED** [review]
Clean code! And great work refactoring.

**Pull request merged**